### PR TITLE
[Improvement]: Add exception if no parent id was set

### LINKED ISSUE #12902: [Improvement]: Add exception if no parent id was set
### Improvement description

- [x] Add a new exception if no parent id was set for dataObjects, assets and documents.
- [x] Improve existing exception message for identical parent id and id
- [x] Add corresponding tests

See #12897

### PR DESCRIPTION
**Pull request by mcop1**
Resolves #12902

### CODE REVIEW

File: models/Asset.php [hunk 1]
```diff
@@ -640,15 +644,6 @@ public function correctPath()
                 // use the parent's path from the database here (getCurrentFullPath), to ensure the path really exists and does not rely on the path
                 // that is currently in the parent asset (in memory), because this might have changed but wasn't not saved
                 $this->setPath(str_replace('//', '/', $parent->getCurrentFullPath() . '/'));
-            } else {
-                trigger_deprecation(
```
Comment (blankse) [inline models/Asset.php:644]: I think we should also throw an exception if the parent with the given ID doesn't exist.

File: models/Asset.php [hunk 2]
```diff
@@ -627,8 +627,12 @@ public function correctPath()
                 throw new Exception("invalid filename '" . $this->getKey() . "' for asset with id [ " . $this->getId() . ' ]');
             }
 
+            if($this->getParentId() === null || $this->getParentId() === 0) {
```
Comment (brusch) [inline models/Asset.php]:
```suggestion
            if(!$this->getParentId()) {
```
Should be sufficient here 😊

File: models/DataObject/AbstractObject.php [hunk 3]
```diff
@@ -852,8 +852,12 @@ protected function correctPath()
                 throw new \Exception('invalid key for object with id [ '.$this->getId().' ] key is: [' . $this->getKey() . ']');
             }
 
+            if($this->getParentId() === null || $this->getParentId() === 0) {
```
Comment (brusch) [inline models/DataObject/AbstractObject.php]:
```suggestion
            if(!$this->getParentId()) {
```

File: models/Document.php [hunk 4]
```diff
@@ -490,24 +490,19 @@ private function correctPath()
                 throw new \Exception('invalid key for document with id [ ' . $this->getId() . ' ] key is: [' . $this->getKey() . ']');
             }
 
+            if($this->getParentId() === null || $this->getParentId() === 0) {
```
Comment (brusch) [inline models/Document.php]:
```suggestion
            if(!$this->getParentId()) {
```

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Asset.php (hunk 1 above)
Comment (brusch) [reply models/Asset.php:644]: Agree, we need to throw an exception here instead 😊

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Asset.php (hunk 1 above)
Comment (mcop1) [reply models/Asset.php:644]: You are right, added the exception + tests

**Pull request merged**